### PR TITLE
Platform blink

### PR DESCRIPTION
**Pull request by Gavin-L26**
Implemented Platform blink to signify the player to jump soon:
1. The blink starts 0.5 seconds before the jump command time, and blinks 2 times. (Reason for 0.5 sec is because the latter half of the game have some very fast subsequent jumps, an earlier time to start blink may cause different blinks to overlap and jumbled up)
2. Commented out the MidiManger related codes since it interferes with the platform blink

### CODE REVIEW

File: Assets/Scripts/ManagerScripts/MidiManager.cs [hunk 1]
```diff
@@ -1,118 +1,118 @@
-using System;
```
Comment (jamilfaisal) [inline Assets/Scripts/ManagerScripts/MidiManager.cs:1]: Since this file is commented out, let's move it to the archive folder and remove it from the scene (if it's still there)

File: Assets/Scripts/PlayerScripts/PlayerAction.cs [hunk 2]
```diff
@@ -9,12 +9,20 @@ public class PlayerAction : MonoBehaviour
     public Melanchall.DryWetMidi.MusicTheory.NoteName noteRestriction;
     public KeyCode input;
     public List<double> timeStamps = new List<double>();
+    public double blinkOffset;
```
Comment (jamilfaisal) [inline Assets/Scripts/PlayerScripts/PlayerAction.cs:12]: Any reason why these two attributes are public instead of private?

File: Assets/Scripts/ObjectScripts/NeutralPlatform.cs [hunk 3]
```diff
@@ -3,5 +3,28 @@
...
+        PlatformManager.current.BlinkEvent += Blink;
+    }
+    private void Blink()
+    {
+        StartCoroutine(BlinkDelay());
+    }
+
+    private IEnumerator BlinkDelay()
+    {
+        _material.color = _startColor * 1.5f;
```
Comment (jamilfaisal) [inline Assets/Scripts/ObjectScripts/NeutralPlatform.cs]: 1.5f and 0.125f should be converted into constants

File: Assets/Scripts/PlayerScripts/PlayerAction.cs [hunk 4]
```diff
@@ -40,6 +48,16 @@ private void Update()
             _marginOfError = MusicPlayer.current.marginOfError;
             _audioTime = MusicPlayer.GetAudioSourceTime() - (MusicPlayer.current.inputDelayInMilliseconds / 1000.0);
 
+            if(!_ableToBlink && _audioTime > _previousBlink + blinkCooldown){
```
Comment (jamilfaisal) [inline Assets/Scripts/PlayerScripts/PlayerAction.cs]: Lines 51-59 can be refactored into one method since it's all related to blinking

**Comment by jamilfaisal** [comment]
Base automatically changed to main, make sure to change it to dev

**Review by jamilfaisal — CHANGES_REQUESTED** [review]
I really like this!!! One more minor comment, can we lower the brightness a little bit? It's very shocking the way it is now (like lightning is striking the platforms haha)

**Review by lilvedaes — CHANGES_REQUESTED** [review]
It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.

Basically two changes:
- Add L and R movement
- Make the "hint" blink less bright than the actual action blink

**Comment by Gavin-L26** [comment]
> 

This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump. 
The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap. 

So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.

**Comment by Gavin-L26** [comment]
> It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.
> 
> Basically two changes:
> 
>     * Add L and R movement
> 
>     * Make the "hint" blink less bright than the actual action blink

This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump.
The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap.

So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.

**Comment by lilvedaes** [comment]
> > It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.
> > Basically two changes:
> > ```
> > * Add L and R movement
> > 
> > * Make the "hint" blink less bright than the actual action blink
> > ```
> 
> This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump. The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap.
> 
> So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.

What if we make the blinks colour different depending on the action? and then we could do first blink -> brighter blink on action beat

**Comment by Gavin-L26** [comment]
> > > It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.
> > > Basically two changes:
> > > ```
> > > * Add L and R movement
> > > 
> > > * Make the "hint" blink less bright than the actual action blink
> > > ```
> > 
> > 
> > This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump. The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap.
> > So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.
> 
> What if we make the blinks colour different depending on the action? and then we could do first blink -> brighter blink on action beat

I don't think that solves the overlapping blinking problem, maybe I didn't explain it more clearly.
Let's say the jump blink is green and left blink is yellow: 

Jump:   . . . - ......... - blink - blink - jump
                                 ^
Left:      . . . - blink - blink - left   - ...........
                                 ^

What colour should this blink be? Green, yellow, or a mix of both?

**Comment by Gavin-L26** [comment]
> > > It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.
> > > Basically two changes:
> > > ```
> > > * Add L and R movement
> > > 
> > > * Make the "hint" blink less bright than the actual action blink
> > > ```
> > 
> > 
> > This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump. The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap.
> > So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.
> 
> What if we make the blinks colour different depending on the action? and then we could do first blink -> brighter blink on action beat

Sorry the diagram didn't work after I posted the comment. Here is an image: 
image

What colour should this blink be? Green, yellow, or a mix of both?

**Comment by lilvedaes** [comment]
> > > > It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.
> > > > Basically two changes:
> > > > ```
> > > > * Add L and R movement
> > > > 
> > > > * Make the "hint" blink less bright than the actual action blink
> > > > ```
> > > 
> > > 
> > > This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump. The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap.
> > > So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.
> > 
> > 
> > What if we make the blinks colour different depending on the action? and then we could do first blink -> brighter blink on action beat
> 
> Sorry the diagram didn't work after I posted the comment. Here is an image: image
> 
> What colour should this blink be? Green, yellow, or a mix of both?

Ahh I see what you mean, what if we only give it one previous blink to the action and then the action blink? And we make these two blinks closer between each other than the actions are. Like:
Blink —- Action Blink

then if we have two actions followed after the other it would be:

Blink - Jump Blink
———————————-Blink - Left Blink

so they don’t overlap

**Comment by Gavin-L26** [comment]
> > > > > It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.
> > > > > Basically two changes:
> > > > > ```
> > > > > * Add L and R movement
> > > > > 
> > > > > * Make the "hint" blink less bright than the actual action blink
> > > > > ```
> > > > 
> > > > 
> > > > This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump. The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap.
> > > > So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.
> > > 
> > > 
> > > What if we make the blinks colour different depending on the action? and then we could do first blink -> brighter blink on action beat
> > 
> > 
> > Sorry the diagram didn't work after I posted the comment. Here is an image: image
> > What colour should this blink be? Green, yellow, or a mix of both?
> 
> Ahh I see what you mean, what if we only give it one previous blink to the action and then the action blink? And we make these two blinks closer between each other than the actions are. Like: Blink —- Action Blink
> 
> then if we have two actions followed after the other it would be:
> 
> Blink - Jump Blink ———————————-Blink - Left Blink
> 
> so they don’t overlap

Yeah, I can try that and we can see how it looks

**Comment by lilvedaes** [comment]
> > > > > > It seems the blinking is not activated yet for left and right movement, could you update it so it is? My other issue is that I see two blinks for the jump, I know that the timing issue is related to the MIDI but why do we have two blinks? Is the first one a warning and the second one the real one? If so, that's better than having one blink, but I would like the first blink to be less bright than the second one to make it clear that the first one is a lead to the second one.
> > > > > > Basically two changes:
> > > > > > ```
> > > > > > * Add L and R movement
> > > > > > 
> > > > > > * Make the "hint" blink less bright than the actual action blink
> > > > > > ```
> > > > > 
> > > > > 
> > > > > This was an old version of platform blink where I basically used blinking as a way for the user to get ready for the jump. The idea is that the platform blinks for the two beat before the jump and on the third beat the player jump (sort of like "blink - blink - jump"). But this is going to be difficult with side movement added too, for example if a side movement is required one beat before a jump, the blink warning for the two are going to overlap.
> > > > > So if we want the blink to warn for all actions instead of just jump, I think the best way is to have the platform only blink once right on the beat when an action is required.
> > > > 
> > > > 
> > > > What if we make the blinks colour different depending on the action? and then we could do first blink -> brighter blink on action beat
> > > 
> > > 
> > > Sorry the diagram didn't work after I posted the comment. Here is an image: image
> > > What colour should this blink be? Green, yellow, or a mix of both?
> > 
> > 
> > Ahh I see what you mean, what if we only give it one previous blink to the action and then the action blink? And we make these two blinks closer between each other than the actions are. Like: Blink —- Action Blink
> > then if we have two actions followed after the other it would be:
> > Blink - Jump Blink ———————————-Blink - Left Blink
> > so they don’t overlap
> 
> Yeah, I can try that and we can see how it looks

awesome thanks! to be clearer, an example would be if you have a jump half a second after you have a left movement then it would look like:

0.25 ----------0.5----------0.75----------1--
Pre Blink - Blink Left ------------------------
-------------------------Pre Blink - Blink Jump

Or if you figure out a better way that's also okay c:

File: Assets/Scripts/PlayerScripts/PlayerAction.cs (hunk 2 above)
Comment (Gavin-L26) [reply Assets/Scripts/PlayerScripts/PlayerAction.cs:12]: I edit this offset value on the inspector screen, that's the only reason it is public

**Comment by Gavin-L26** [comment]
A new version of the platform blink is implemented. It is now less of a blink and more of a colour change (blinking various colour very fast doesn't look very good). The platform changes to a darker version of the action colour 0.5 sec before the jump and changes to a lighter version right at the jump time.

Action Colour:
Jump (Green)
Left (Yellow)
Right (Purple)

**Review by lilvedaes — COMMENTED** [review]
The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.

**Comment by Gavin-L26** [comment]
> The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.

Can you explain what is weird about them? I'm not sure what to fix with them being weird.

**Comment by lilvedaes** [comment]
> > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> 
> Can explain what is weird about them? I'm not sure what to fix with them being weird.

They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.

**Comment by lilvedaes** [comment]
> > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > 
> > 
> > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> 
> They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.

Also in some parts there is no blink for when you have to move left either, you'll see on the video I'm about to send on discord.

**Comment by Gavin-L26** [comment]
> > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > 
> > 
> > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> 
> They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.

\

> > > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > > 
> > > 
> > > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> > 
> > 
> > They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.
> 
> Also in some parts there is no blink for when you have to move left either, you'll see on the video I'm about to send on discord.

I'm pretty confident all the timing for the actions are correct, so I'm pretty sure not all left - right motion has an instruction in the midifile

**Comment by lilvedaes** [comment]
> > > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > > 
> > > 
> > > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> > 
> > 
> > They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.
> 
> \
> 
> > > > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > > > 
> > > > 
> > > > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> > > 
> > > 
> > > They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.
> > 
> > 
> > Also in some parts there is no blink for when you have to move left either, you'll see on the video I'm about to send on discord.
> 
> I'm pretty confident all the timing for the actions are correct, so I'm pretty sure not all left - right motion has an instruction in the midifile

ohh so it's a MIDI issue then? we have to update it so all the L and R instructions are in it. Is that something Andrew is working on? I would still like to test this with the latest updated MIDI

**Comment by Gavin-L26** [comment]
> > > > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > > > 
> > > > 
> > > > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> > > 
> > > 
> > > They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.
> > 
> > 
> > \
> > > > > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > > > > 
> > > > > 
> > > > > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> > > > 
> > > > 
> > > > They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.
> > > 
> > > 
> > > Also in some parts there is no blink for when you have to move left either, you'll see on the video I'm about to send on discord.
> > 
> > 
> > I'm pretty confident all the timing for the actions are correct, so I'm pretty sure not all left - right motion has an instruction in the midifile
> 
> ohh so it's a MIDI issue then? we have to update it so all the L and R instructions are in it. Is that something Andrew is working on? I would still like to test this with the latest updated MIDI

I just grabbed the MIDI file and checked its content on a MIDI viewer, it seems they do have instructions. However, many side movement instructions are very close together (within 0.25 sec) which is basically in the cooldown time I've set. This means the entire colour change sequence needs to well within 0.25 secs. So two blinks might not work.

**Comment by lilvedaes** [comment]
> > > > > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > > > > 
> > > > > 
> > > > > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> > > > 
> > > > 
> > > > They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.
> > > 
> > > 
> > > \
> > > > > > > The jump looks great! Could make the second blink a bit brighter? Also, the left and right blinks look a bit weird, is this because of the un-updated midi file? If so, let me know when it's updated to test with that, otherwise, for today we can merge it and disable the L and R links if they still look weird by the deadline time.
> > > > > > 
> > > > > > 
> > > > > > Can explain what is weird about them? I'm not sure what to fix with them being weird.
> > > > > 
> > > > > 
> > > > > They turn on when I don't have to turn right or left. I only saw one purple blink which wouldn't make sense because every time you go left you eventually go right. And the blinks were kind of tangled between each other. Let me see if I can take a video.
> > > > 
> > > > 
> > > > Also in some parts there is no blink for when you have to move left either, you'll see on the video I'm about to send on discord.
> > > 
> > > 
> > > I'm pretty confident all the timing for the actions are correct, so I'm pretty sure not all left - right motion has an instruction in the midifile
> > 
> > 
> > ohh so it's a MIDI issue then? we have to update it so all the L and R instructions are in it. Is that something Andrew is working on? I would still like to test this with the latest updated MIDI
> 
> I just grabbed the MIDI file and checked its content on a MIDI viewer, it seems they do have instructions. However, many side movement instructions are very close together (within 0.25 sec) which is basically in the cooldown time I've set. This means the entire colour change sequence needs to well within 0.25 secs. So two blinks might not work.

Can we make the blinks closer to each other? Or try having two blinks for jump and one blink for L and R?

**Review by lilvedaes — CHANGES_REQUESTED** [review]
Could we only have the reduced gap for left and right? The one for jump was good. Maybe when you have two notes at the same time and one is jump, only display the jump blink. Also at the beginning of the level a lot of the jump blinks are not showing. Not sure if its a MIDI issue or not. Could you merge dev into your branch to see how it looks like with the updated fixes? to avoid any merging issues just do:
git checkout dev
git pull
git checkout Platform-blink
git merge dev

**Review by lilvedaes — APPROVED** [review]
Nice! It looks awesome c:

**Comment by lilvedaes** [comment]
> Nice! It looks awesome c:

x2, both jump and L/R blink still show at diagonal but they don't look like a glitch anymore, they come one after the other, I reapprove. Can be improved after but I say its worth the merge.

File: Assets/Scripts/PlayerScripts/PlayerAction.cs (hunk 2 above)
Comment (jamilfaisal) [reply Assets/Scripts/PlayerScripts/PlayerAction.cs:12]: It's a constant value though, right? Let's try to keep inspector screen assignments only for things that are necessary (assignments to other game objects for example) so we don't clutter it

File: Assets/Scripts/UIScripts/PauseScreen.cs [hunk 5]
```diff
@@ -43,8 +43,8 @@ public void Resume()
     {
         pauseMenuUI.SetActive(false);
         settingsMenuUI.SetActive(false);
-        MidiManager.current.ResumePlayback();
-        if (Time.timeSinceLevelLoad > 5)
+        //MidiManager.current.ResumePlayback();
+        if (Time.time > 5)
```
Comment (jamilfaisal) [inline Assets/Scripts/UIScripts/PauseScreen.cs:47]: Why is this change necessary? I believe @judy-n changed this to timeSinceLevelLoad to avoid issues when playtesting with a build (instead of the unity editor) (since Time.time does not reset to 0 when a scene loads/reloads)

Comment (judy-n) [reply Assets/Scripts/UIScripts/PauseScreen.cs:47]: yeah exactly, everywhere that uses `Time.time > 5` should be `Time.timeSinceLevelLoad > 5` . It fixes a bug we had, like if you waited 5 seconds in the main menu screen then this wait will never happen unless you use "timeSinceLevelLoad"
